src/plugins/displayLayout/components/DisplayLayout.vue

```diff
@@ -226,18 +227,61 @@
             addItem(itemType, ...options) {
                 let item = getItemDefinition(itemType, this.openmct, this.gridSize, ...options);
                 item.type = itemType;
+                item.id = uuid();
                 this.trackItem(item);
                 this.layoutItems.push(item);
                 this.openmct.objects.mutate(this.internalDomainObject, "configuration.items", this.layoutItems);
                 this.initSelectIndex = this.layoutItems.length - 1;
             },
             trackItem(item) {
                 if (item.type === "telemetry-view") {
-                    this.telemetryViewMap[this.openmct.objects.makeKeyString(item.identifier)] = true;
+                    let keyString = this.openmct.objects.makeKeyString(item.identifier);
+                    let count = this.telemetryViewMap[keyString] || 0;
+                    this.telemetryViewMap[keyString] = ++count;
                 } else if (item.type === "subobject-view") {
                     this.objectViewMap[this.openmct.objects.makeKeyString(item.identifier)] = true;
                 }
             },
+            removeItem(item, index) {
+                this.initSelectIndex = -1;
+                this.layoutItems.splice(index, 1);
+                this.mutate("configuration.items", this.layoutItems);
+                this.untrackItem(item);
+                this.$el.click();
+            },
+            untrackItem(item) {
+                if (item.type === 'telemetry-view') {
+                    let keyString = this.openmct.objects.makeKeyString(item.identifier);
+
+                    if (this.telemetryViewMap[keyString]) {
+                        let count = --this.telemetryViewMap[keyString];
+
+                        if (count === 0) {
+                            delete this.telemetryViewMap[keyString];
+                            this.removeFromComposition(keyString);
+                        }
+                    }
+                } else if (item.type === 'subobject-view') {
+                    let keyString = this.openmct.objects.makeKeyString(item.identifier);
+
+                    if (this.objectViewMap[keyString]) {
+                        delete this.objectViewMap[keyString];
+                        this.removeFromComposition(keyString);
+                    }
+                }
+            },
+            removeFromComposition(keyString) {
+                let composition = _.get(this.internalDomainObject, 'composition');
+                composition = composition.filter(identifier => {
+                    let childKeyString = this.openmct.objects.makeKeyString(identifier);
+                    if (childKeyString === keyString && this.telemetryViewMap[keyString]) {
```

Comment:
This conditional is unnecessary as untrackItem already does a similar check, and removeFromComposition is only called from untrackItem.